transformers/storage/pot/repository_test.go

```diff
@@ -0,0 +1,210 @@
+			userPieMetadata := storage.GetValueMetadata(pot.UserPie, map[storage.Key]string{constants.MsgSender: fakeAddress}, storage.Uint256)
+			setupErr := repo.Create(diffID, fakeHeaderID, userPieMetadata, fakeUint256)
+			Expect(setupErr).NotTo(HaveOccurred())
+
+			insertErr := repo.Create(diffID, fakeHeaderID, userPieMetadata, fakeUint256)
+			Expect(insertErr).NotTo(HaveOccurred())
+
+			Expect(insertErr).NotTo(HaveOccurred())
+			var count int
+			getCountErr := db.Get(&count, `SELECT COUNT(*) FROM maker.pot_user_pie`)
+			Expect(getCountErr).NotTo(HaveOccurred())
+			Expect(count).To(Equal(1))
+		})
+
+		It("returns an error if metadata missing ilk", func() {
+			malformedUserPieMetadata := storage.GetValueMetadata(pot.UserPie, nil, storage.Uint256)
+
+			err := repo.Create(diffID, fakeHeaderID, malformedUserPieMetadata, fakeUint256)
+			Expect(err).To(MatchError(storage.ErrMetadataMalformed{MissingData: constants.MsgSender}))
+		})
+	})
+
+	Describe("Pie", func() {
+		inputs := shared_behaviors.StorageBehaviorInputs{
+			ValueFieldName:   pot.Pie,
+			Value:            fakeUint256,
+			StorageTableName: "maker.pot_pie",
+			Repository:       &repo,
+			Metadata:         pot.PieMetadata,
+		}
+
+		shared_behaviors.SharedStorageRepositoryBehaviors(&inputs)
+	})
+
+	Describe("dsr", func() {
+		inputs := shared_behaviors.StorageBehaviorInputs{
+			ValueFieldName:   pot.Dsr,
+			Value:            fakeUint256,
+			StorageTableName: "maker.pot_dsr",
+			Repository:       &repo,
+			Metadata:         pot.DsrMetadata,
+		}
+
+		shared_behaviors.SharedStorageRepositoryBehaviors(&inputs)
+	})
+
+	Describe("chi", func() {
+		inputs := shared_behaviors.StorageBehaviorInputs{
+			ValueFieldName:   pot.Chi,
+			Value:            fakeUint256,
+			StorageTableName: "maker.pot_chi",
+			Repository:       &repo,
+			Metadata:         pot.ChiMetadata,
+		}
+
+		shared_behaviors.SharedStorageRepositoryBehaviors(&inputs)
+	})
+
+	Describe("vat", func() {
+		It("persists a record", func() {
+			err := repo.Create(diffID, fakeHeaderID, pot.VatMetadata, fakeAddress)
+			Expect(err).NotTo(HaveOccurred())
+
+			addressID, addressErr := shared.GetOrCreateAddress(fakeAddress, db)
+			Expect(addressErr).NotTo(HaveOccurred())
+			var result VariableRes
+			err = db.Get(&result, `SELECT diff_id, header_id, vat AS value FROM maker.pot_vat`)
+			Expect(err).NotTo(HaveOccurred())
+			AssertVariable(result, diffID, fakeHeaderID, strconv.FormatInt(addressID, 10))
+		})
+
+		It("doesn't duplicate a record", func() {
+			setupErr := repo.Create(diffID, fakeHeaderID, pot.VatMetadata, fakeAddress)
+			Expect(setupErr).NotTo(HaveOccurred())
+
+			createErr := repo.Create(diffID, fakeHeaderID, pot.VatMetadata, fakeAddress)
+			Expect(createErr).NotTo(HaveOccurred())
+
+			var count int
+			query := `SELECT COUNT(*) FROM maker.pot_vat`
+			queryErr := db.Get(&count, query)
+			Expect(queryErr).NotTo(HaveOccurred())
+			Expect(count).To(Equal(1))
+		})
```

Comment:
Why not uses the shared behaviors here?

Reply:
The "persists a record" test is more complex than the shared test suite, because it has to deal with the address FK. But the "duplicate record" test is identical to the one in the shared suite, so maybe it would be worth breaking apart the shared test suite to allow us to extract only the "duplicate record" one in cases like these.

Reply:
ah, that makes sense - would be interested in refactors that make it easier to use the shared tests for fields including a FK, but definitely doesn't need to be a part of this PR 👍